test/integration/garbagecollector/garbage_collector_test.go

```diff
@@ -242,8 +242,9 @@ func setupWithServer(t *testing.T, result *kubeapiservertesting.TestServer, work
 	if err != nil {
 		t.Fatalf("failed to create dynamicClient: %v", err)
 	}
-	sharedInformers := informers.NewSharedInformerFactory(clientSet, 0)
-	metadataInformers := metadatainformer.NewSharedInformerFactory(metadataClient, 0)
+	onListError := func(error) bool { return true }
```

Comment:
> Here you go @caesarxuchao this is the log of the failing integration test using the new informer shutdown mechanism
> [test-crd-deletion-failing.log](https://github.com/kevindelgado/kubernetes/files/5632956/test-crd-deletion-failing.log)
> as well as the log of the passing test using the old method
> [test-crd-deletion-passing.log](https://github.com/kevindelgado/kubernetes/files/5632957/test-crd-deletion-passing.log)

I realized I didn't add any context to the logs I posted, but basically the only difference between the two is that "passing" has onListError return false (simulating the old behavior of never shutting down erring informers) while "failing" has it set to true as seen here.

Reply:
I saw this error:

```
garbage_collector_test.go:1285: failed waiting for dependent "dependentzsfcv" (owned by "ownervmvn8") to be deleted
```

Do you know if this was emitted during the first pass or the second pass of CRD cascading deletion?